src/wasm/wasm-validator.cpp

```diff
@@ -210,6 +210,11 @@ struct FunctionValidator : public WalkerPass<PostWalker<FunctionValidator>> {
 
   std::unordered_map<Name, BreakInfo> breakInfos;
 
+  std::vector<Expression*> ancestors;
```

Comment:
instead of manually doing this, there is `ControlFlowWalker` and `ExpressionStackWalker`.

Reply:
Awesome, I have used `ExpressionStackWalker` extensively now.